src/pages/Dashboard/components/DashboardHeader/MobileUserHeader.tsx

```diff
@@ -0,0 +1,73 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import { Box, Flex, IconButton, Text } from '@chakra-ui/react'
+import { useMemo } from 'react'
+import { FaExpand } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+
+import { ProfileAvatar } from '../ProfileAvatar/ProfileAvatar'
+
+import { useWallet } from '@/hooks/useWallet/useWallet'
+import { useMipdProviders } from '@/lib/mipd'
+import { selectWalletRdns } from '@/state/slices/localWalletSlice/selectors'
+import { useAppSelector } from '@/state/store'
+
+const searchIcon = <SearchIcon />
+const qrCodeIcon = (
+  <Box>
+    <FaExpand />
+  </Box>
+)
+
+const mobileButtonRowDisplay = { base: 'flex', md: 'none' }
+
+type MobileUserHeaderPrps = {
+  onSearchOpen: () => void
+  handleQrCodeClick: () => void
+  onOpen: () => void
+}
+
+export const MobileUserHeader = ({
+  onSearchOpen,
+  handleQrCodeClick,
+  onOpen,
+}: MobileUserHeaderPrps) => {
+  const translate = useTranslate()
+
+  const {
+    state: { walletInfo },
+  } = useWallet()
+
+  const maybeRdns = useAppSelector(selectWalletRdns)
+
+  const mipdProviders = useMipdProviders()
+  const maybeMipdProvider = useMemo(
+    () => mipdProviders.find(provider => provider.info.rdns === maybeRdns),
+    [mipdProviders, maybeRdns],
+  )
+
+  return (
+    <Flex justifyContent='space-between' width='100%' display={mobileButtonRowDisplay}>
+      <Flex align='center' onClick={onOpen}>
+        <ProfileAvatar size='md' borderRadius='full' />
+        <Text ml={2} fontWeight='semibold' fontSize='md'>
+          {(maybeMipdProvider?.info?.name || walletInfo?.meta?.label || walletInfo?.name) ??
```

Comment:
Fixes the wallet name for mipd wallets always displaying as "MetaMask" on the new user menu